### PR TITLE
Initial integration tests

### PR DESCRIPTION
**Pull request by nitro-marky**
Introduces some initial integration tests which cover the `out of bounds red v2` executable.

Some of the expected odds seem unusual, expecting success when it should fail, at the moment but that's how they are currently dealt with in the Morello part.

The tests cover the following areas:

- happy path
- invalid executable (not in list)
- passing parameters
- bad parameters

### CODE REVIEW

File: integration/helper/routeHelper.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import request from 'supertest'
+
+export async function getOutOfBoundsReadAarch64(app: any, password: any, firstOffset: any, secondOffset: any) {
```
Comment (jonmattgray) [inline integration/helper/routeHelper.ts]: it should be possible to give these types e.g. `app: Express, password: string` etc

Comment (nitro-marky) [reply integration/helper/routeHelper.ts]: The app can. The rest can be it depends on how strictly typed we want. I guess we always want a password to be a string, the others could be provided as numbers or strings

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "experimentalDecorators": true,
+    "checkJs": false,
```
Comment (jonmattgray) [inline tsconfig.json]: is this left over from when routeHelper was `.js`?

File: integration/helper/routeHelper.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  password: any,
+  firstOffset: any,
+  secondOffset: any
+) {
+  return request(app)
+    .get(`/scenario/out-of-bounds-readV2-aarch64?params=${password}&params=${firstOffset}&params=${secondOffset}`)
+    .set('Accept', 'application/json')
+    .set('Content-Type', 'application/json')
+    .send()
+    .then((response: any) => {
```
Comment (jonmattgray) [inline integration/helper/routeHelper.ts]: I think the `any`s on response and err can be removed

File: tsconfig.json (hunk 2 above)
Comment (nitro-marky) [reply tsconfig.json]: Yeah that can go now.

File: integration/helper/routeHelper.ts (hunk 3 above)
Comment (nitro-marky) [reply integration/helper/routeHelper.ts]: Yeah, weirdly when I first wrote it it enforced types on it, now it's stopped, strange

File: integration/helper/routeHelper.ts (hunk 1 above)
Comment (jonmattgray) [reply integration/helper/routeHelper.ts]: you can do `firstOffset: number | string`, just trying to remove all `any`

File: integration/out-of-bounds-read.test.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    expect(response.status).to.equal(200)
+    expect(response.body.status).to.contain('success')
+  })
+
+  test('Bad Parameters - aarch64', async () => {
+    const response = await getOutOfBoundsReadAarch64(app, 'badpass', NaN, 'ttttttt')
+
+    expect(response.status).to.equal(200)
+    expect(response.body.status).to.contain('success')
```
Comment (jonmattgray) [inline integration/out-of-bounds-read.test.ts:30]: I see the response body for this test and the `Missing parameters...` test is ` 'Storing Secret...\nThe data between indices 0 and 0 is: \n'`. Is that the kind of output we expect from bad params and do we want to assert that output in these tests?

File: integration/out-of-bounds-read.test.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+describe('Tests aarch64 version', () => {
+  let app: express.Express
+
+  before(async function () {
+    app = await CreateHttpServer()
+  })
+
+  test('Happy Path - aarch64', async () => {
+    const response = await getOutOfBoundsReadAarch64(app, 'pass', -32, -28)
+
```
Comment (jonmattgray) [inline integration/out-of-bounds-read.test.ts:20]: maybe assert `response.body.output` is revealing the password `Storing Secret...\nThe data between indices -32 and -28 is: pass\n'`

File: integration/out-of-bounds-read.test.ts (hunk 4 above)
Comment (nitro-marky) [reply integration/out-of-bounds-read.test.ts:30]: I asked Matt about this and apparently at the moment it's how it is. But yeah, I don't get it either.

**Review by n3op2 — APPROVED** [review]

**Pull request merged**